internal/core/src/index/VectorMemIndex.cpp

```diff
@@ -92,18 +92,19 @@ VectorMemIndex<T>::VectorIterators(const milvus::DatasetPtr dataset,
 }
 
 template <typename T>
-BinarySet
+CreateIndexResultPtr
 VectorMemIndex<T>::Upload(const Config& config) {
     auto binary_set = Serialize(config);
     file_manager_->AddFile(binary_set);
 
     auto remote_paths_to_size = file_manager_->GetRemotePathsToFileSize();
-    BinarySet ret;
+    std::vector<SerializedIndexFileInfo> index_files;
```

Comment:
Utilize a common method to create index file vectors from remote files , reducing code duplication.